src/m365/planner/commands/task/task-reference-add.ts

```diff
@@ -0,0 +1,104 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { formatting } from '../../../../utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  taskId: string;
+  url: string;
+  alias?: string;
+  type?: string;
+}
+
+class PlannerTaskReferenceAddCommand extends GraphCommand {  
+  public get name(): string {
+    return commands.TASK_REFERENCE_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new reference to a Planner task';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getTaskDetailsEtag(args.options.taskId)
+      .then(etag => {
+        const requestOptionsTaskDetails: any = {
+          url: `${this.resource}/v1.0/planner/tasks/${args.options.taskId}/details`,
```

Comment:
just noticed, shouldn't we `${encodeURIComponent(args.options.taskId)}` here 🤔

Reply:
I will add it in my review by myself. No actions required